site/content/docs/developing/publish-subscribe.ja.md

```diff
@@ -0,0 +1,86 @@
+# パブリッシュ / サブスクライブ アーキテクチャ
+
+Copilot [Worker Services](../manifest/worker-service.ja.md)は、すべてのサービスタイプとジョブタイプに共通する `publish` フィールドを利用して、サービス間でメッセージを受け渡すためのパブリッシュ / サブスクライブロジックを簡単に作成することができます。
+
+AWS の一般的なパターンは、メッセージの配信と処理を行うための SNS と SQS の組み合わせです。[SNS](https://docs.aws.amazon.com/ja_jp/sns/latest/dg/welcome.html) は堅牢なメッセージ配信システムで、メッセージの配信を保証しながら複数のサブスクライブしたエンドポイントにメッセージを送ることができます。
+
+[SQS](https://docs.aws.amazon.com/ja_jp/AWSSimpleQueueService/latest/SQSDeveloperGuide/welcome.html)は、メッセージの非同期処理を可能にするメッセージキューです。キューには 1 つまたは複数の SNS トピックや、AWS EventBridge からのイベントを投入をすることができます。
+
+この 2 つのサービスを組み合わせることで、メッセージの送受信を効果的に切り離すことができます。つまり、パブリッシャーは自分のトピックをサブスクライブしているキューを意識する必要がなく、また Worker Service のコードはメッセージがどこから来るかを気にする必要がありません。
+
+## パブリッシャーからのメッセージ送信
+
+既存のサービスから SNS へのメッセージのパブリッシュを許可するには、Manifest に `publish` フィールドを設定するだけです。トピックの機能を表す名前と、このトピックをサブスクライブする権限を付与する Worker Service のオプションリストが必要です。これらのサービスは、パブリッシャーの Manifest に書いた時点では必ずしも存在している必要はありません。
+
+```yaml
+# api サービス用の manifest.yml
+name: api
+type: Backend Service
+
+publish:
+  topics:
+    - name: orders
+      allowed_workers: [orders-worker, receipts-worker]
+```
+
+これにより、[SNS トピック](https://docs.aws.amazon.com/ja_jp/sns/latest/dg/welcome.html)が作成されます。また、トピックにリソースポリシーが設定され、`orders-worker` と `receipts-worker` という Copilot Service がサブスクリプションを作成できるようになります。
+
+また Copilot は、任意の SNS トピックの ARN をコンテナ内の環境変数 `COPILOT_SNS_TOPIC_ARNS` に注入します。
+
+### Javascript での例
+パブリッシャーのサービスがデプロイされると、AWS SDK for SNS を介して SNS にメッセージを送信できるようになります。
```

Comment:
There is no "AWS SDK for SNS" actually, but I understand you chose to keep it as-is 😇